src/main/kotlin/interactor/GetSalaryAverageAndCitiesNamesInCountryInteractor.kt

```diff
@@ -0,0 +1,28 @@
+package interactor
+
+import model.CityEntity
+import java.io.IOException
+
+class GetSalaryAverageAndCitiesNamesInCountryInteractor(
+    private val dataSource: CostOfLivingDataSource
+    ) {
+
+    fun execute(countryName:String):List<Pair<String,Float?>> {
+        return if (checkIfCountryIsValid(countryName))
+            dataSource.getAllCitiesData()
+                .filter{excludeNullSalariesAndLowQualityData(it) && it.country.lowercase() == countryName.lowercase()}
```

Comment:
Execlude Low quality data and null salaries first, then check if the country name is valid, and you should make the **checkIfCountryIsValid** in the filter method
```suggestion
          dataSource.getAllCitiesData()
          .filter{excludeNullSalariesAndLowQualityData(it) && checkIfCountryIsValid(it.countryName)}
```

Reply:
I tried this but it gives me wrong tests

Reply:
No, it's okay, I'm sorry, your code is more efficient

Reply:
First, it checks if the country name is valid, then and ONLY then should continue.